polar-c-api/src/lib.rs

```diff
@@ -1,12 +1,49 @@
+// everything in this file is unsafe, clippy
+#![allow(clippy::not_unsafe_ptr_arg_deref)]
+
+use polar_core::error::PolarError;
 pub use polar_core::polar::Polar;
 pub use polar_core::query::Query;
 use polar_core::{error, terms};
 
-use std::ffi::{CStr, CString};
+use std::ffi::{c_void, CStr, CString};
 use std::os::raw::c_char;
 use std::panic::{catch_unwind, AssertUnwindSafe};
 use std::ptr::{null, null_mut};
-use std::sync::Mutex;
+
+/// Wrapper struct to help us return errors
+#[repr(C)]
+pub struct CResult<T> {
+    pub result: *mut T,
```

Comment:
Short answer is because we are passing ownership of the memory back to the host. E.g. when you create a query. I doubt it actually changes anything since no other language will do anything with that information?